### PR TITLE
Add gatling benchmarks

### PR DESCRIPTION
**Pull request by Aschen**
## What does this PR do ?

Add a Gatling benchmarking suite.

Results are available in `gatling/results`.

The benchmark will spawn n `clients` in `duration`. For example `-c 60 -d 60` will spawn 60 clients in 60 seconds.

There is 4 scenarios:

#### HTTP Write Document

 - Protocol: `HTTP`
 - Authenticated: `yes`
 - Action: `document:create`

#### HTTP Get Document

 - Protocol: `HTTP`
 - Authenticated: `yes`
 - Action: `document:get`

#### Websocket Write Document

 - Protocol: `WS`
 - Authenticated: `yes`
 - Action: `document:create`

#### Websocket Get Document

 - Protocol: `WS`
 - Authenticated: `yes`
 - Action: `document:get`

### CODE REVIEW

File: gatling/commands.rb [hunk 1]
```diff
@@ -0,0 +1,122 @@
+require 'optparse'
+require 'net/http'
+require 'json'
+
+class Babysitter
```
Comment (scottinet) [inline gatling/commands.rb:5]: I see what you did there

**Comment by scottinet** [comment]
Just a thought: why do you use Ruby for command parsing, when the project is written in Node.js, which can already easily parse commands using, for instance, [yargs](https://www.npmjs.com/package/yargs)?

I'm not exactly fond of adding new layers of languages (and, therefore, new requirements), when there is no added value to it.

**Review by scottinet — CHANGES_REQUESTED** [review]
It's not clear to me if the `results/*` files are benchmarks examples or mistakenly added to the project: 
in one hand I can see how these files can be useful but, in the other hand, I find no reference to those in the PR description nor in the README file.

In either cases, the gatling result directory should be put in the `.gitignore` file. And if the .html files in this PR really are examples, then those should be put in another, more explicitly named directory.

**Comment by Aschen** [comment]
@Yoann-Abbes PR is much more complete

**Pull request closed without merging**